manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/ConnectorWorker.java

```diff
@@ -0,0 +1,206 @@
+        }
+        if (status.getState() == ConnectorState.READY) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("Managed Connector for '%s' [%s] is ready.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+            }
+
+            // Connector is ready. We can proceed with the deployment of the Processor in the Shard
+            // The Processor will be provisioned by the Shard when it is in ACCEPTED state *and* Connectors are READY (or null).
+            return setReady(connectorEntity);
+        }
+
+        if (status.getState() == ConnectorState.FAILED) {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("Creating Managed Connector for '%s' [%s] failed.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId()));
+            }
+
+            // Deployment of the Connector has failed. Bubble FAILED state up to ProcessorWorker.
+            return setStatus(connectorEntity, ManagedResourceStatus.FAILED);
+        }
+
+        return connectorEntity;
+    }
+
+    private ConnectorEntity deployConnector(ConnectorEntity connectorEntity) {
+        // Creation is performed asynchronously. The returned Connector is a place-holder.
+        Connector connector = connectorsApi.createConnector(connectorEntity);
+        return setConnectorExternalId(connectorEntity, connector.getId());
+    }
+
+    @Transactional
+    protected ConnectorEntity setConnectorExternalId(ConnectorEntity connectorEntity, String connectorExternalId) {
+        ConnectorEntity resource = getDao().findById(connectorEntity.getId());
+        resource.setConnectorExternalId(connectorExternalId);
+        return resource;
+    }
+
+    @Override
+    protected ConnectorEntity runDeleteOfDependencies(Work work, ConnectorEntity connectorEntity) {
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(String.format("Destroying dependencies for '%s' [%s]",
+                    connectorEntity.getName(),
+                    connectorEntity.getId()));
+        }
+
+        // This is idempotent as it gets overridden later depending on actual state
+        connectorEntity = setStatus(connectorEntity, ManagedResourceStatus.DELETING);
+
+        Connector connector = connectorsApi.getConnector(connectorEntity);
+
+        // Step 1 - Clean up Kafka Topic if Connector cannot be found
+        if (Objects.isNull(connector)) {
+            return deleteTopic(connectorEntity);
+        }
+
+        // Step 1 - Clean up Kafka Topic if Connector is DELETED
+        ConnectorStatusStatus status = connector.getStatus();
+        if (Objects.isNull(status)) {
+            return connectorEntity;
+        }
+        if (status.getState() == ConnectorState.DELETED) {
+            return deleteTopic(connectorEntity);
+        }
+        if (status.getState() == ConnectorState.FAILED) {
```

Comment:
If the `Connector` failed to deploy when we created the `Processor`, do this mean we never try to delete it?

I think the logic here needs to be something like: "If the Processor is `FAILED`, then try and delete it and if it doesn't enter the `DELETING` status, something is really wrong"

Reply:
Review note: TODO!

Reply:
I have revised the logic a little to ensure we complete _our_ tidy-up even if MC failed to delete a `Connector`.

IDK if there was some confusion over our `ConnectorEntity` and MC's `Connector`. When we request a `Processor` is deleted we set both it and its `ConnectorEntity` statuses to `DEPROVISION` and invoke this worker mechanism. We first request deletion of the MC `Connector` and enter our polling loop since it's an async operation.

If it is "completely successful" i.e. getConnector returns `null` we continue on to deleting the Kafka Topic and deleting the `ConnectorEntity` database record. If is is "sort of successful" i.e. we get a status from MC of either `DELETED` or `FAILED` we also take this as successful for our purpose and move on to deleting the Kafka Topic and deleting the `ConnectorEntity` database record.